packages/styling/src/styles/Variants.ts

```diff
@@ -0,0 +1,139 @@
+    // themeLight: '#c7e0f4',
+    // themeLighter: '#deecf9',
+    // themeLighterAlt: '#eff6fc',
+
+    // foregrounds
+    // black: '#000000',
+    // neutralDark: '#212121',
+    // neutralPrimary: '#333333',
+    // neutralPrimaryAlt: '#3c3c3c',
+    // neutralSecondary: '#666666',
+    // neutralTertiary: '#a6a6a6',
+
+    // backgrounds
+    // neutralTertiaryAlt: '#c8c8c8', // themeLighter?
+    // neutralQuaternary: '#d0d0d0',
+    // neutralQuaternaryAlt: '#dadada',
+    // neutralLight: '#eaeaea',
+    // neutralLighter: '#f4f4f4',
+    // neutralLighterAlt: '#f8f8f8',
+    white: p.themeLighterAlt
+  };
+
+  let partialSemantic: Partial<ISemanticColors> = {
+    bodyBackground: p.themeLighterAlt,
+
+    inputBorder: 'transparent',
+    // inputBorderHovered: p.neutralPrimary,
+    inputBackground: p.themeLighter,
+    // inputBackgroundChecked: p.themePrimary,
+    // inputBackgroundCheckedHovered: p.themeDarkAlt,
+    inputForegroundChecked: p.themeLighterAlt,
+    // inputFocusBorderAlt: p.themePrimary,
+  };
+
+  return createTheme({ ...theme, ...{ palette: partialPalette, semanticColors: partialSemantic } });
+}
+
+/**
+ * A variant where the background is a strong version of the primary color. All colors change.
+ * The background becomes shades of the primary color.
+ * The foreground/text becomes shades of the background color.
+ * The primary color becomes shades of the background.
+ *
+ * @export
+ * @param {IPartialTheme} theme the theme for which to build a variant for
+ * @returns {ITheme} the variant theme
+ */
+export function getStrongVariant(theme: IPartialTheme): ITheme {
+  let fullTheme = createTheme(theme);
+  let p = fullTheme.palette;
+
+  // dirty algorithm:
+  // in a tricolor theme, foreground doesn't get used?
+  // theme colors -> background shades
+  // foregrounds -> background shades
+  // backgrounds -> theme colors
+  let partialPalette: Partial<IPalette> = {
+    // theme
+    themeDarker: p.white,
+    themeDark: p.neutralLighterAlt,
+    themeDarkAlt: p.neutralLighterAlt,
+    themePrimary: p.white,
+    themeSecondary: p.neutralLighter,
+    themeTertiary: p.neutralLight,
+    themeLight: p.neutralQuaternaryAlt,
+    themeLighter: p.neutralQuaternary,
+    themeLighterAlt: p.neutralTertiaryAlt,
+
+    // foregrounds
+    black: p.neutralLighterAlt,
+    neutralDark: p.neutralLighter,
+    neutralPrimary: p.white,
+    neutralPrimaryAlt: p.neutralLight,
+    neutralSecondary: p.neutralQuaternaryAlt,
+    neutralTertiary: p.neutralQuaternary,
+
+    // backgrounds
+    neutralTertiaryAlt: p.themeLighterAlt,
+    neutralQuaternary: p.themeLighter,
+    neutralQuaternaryAlt: p.themeLight,
+    neutralLight: p.themeTertiary,
+    neutralLighter: p.themeSecondary,
+    neutralLighterAlt: p.themePrimary,
+    white: p.themeDarkAlt
+  };
+
+  let partialSemantic: Partial<ISemanticColors> = {
+    bodyBackground: p.themeDarkAlt,
+    bodyText: p.white,
+
+    inputBorder: 'transparent',
```

Comment:
on second thought this should be inputBackground, just in case some other place uses inputBorder with inputBackground